scripts/remove-duplicate-emails.php

```diff
@@ -6,24 +6,37 @@
  * drush --script-path=../scripts/ php-script remove-duplicate-emails.php
  */
 
-$users = db_query('SELECT uid FROM users 
-  WHERE mail IN (SELECT mail FROM users GROUP BY mail HAVING COUNT(mail) > 1)
-  AND uid NOT IN (SELECT MIN(uid) FROM users GROUP BY mail HAVING COUNT(mail) > 1)');
+$dupes = array_keys(db_query('SELECT mail FROM users GROUP BY mail HAVING COUNT(mail) > 1')->fetchAllKeyed());
+$removed = 0;
 
-foreach ($users as $user) {
-  $user = user_load($user->uid);
+foreach ($dupes as $mail) {
+  // Load all users with that duped email address, with the most recently accessed first.
+  $users = db_query('SELECT uid FROM users WHERE mail = :mail ORDER BY access DESC', [':mail' => $mail]);
+  $canonical_uid = 0;
 
-  if($user->access == 0) {
-    print 'Deleting ' . $user->uid . ' (' . $user->mail . ')' . PHP_EOL;
+  foreach ($users as $key => $user) {
+    $user = user_load($user->uid);
 
-    $northstar_id = $user->field_northstar_id[LANGUAGE_NONE][0]['value'];
+    if ($key == 0) {
```

Comment:
Hmm, I think we need it in the query so we can store the canonical UID on the duped accounts (for the `duplicate-12345-1@dosomething.invalid` placeholder email). Is there a way to get the UID but skip the rest of the record?